platform/linux-generic/odp_packet.c

```diff
@@ -92,12 +92,13 @@ static inline uint32_t packet_seg_len(odp_packet_hdr_t *pkt_hdr,
 	return pkt_hdr->buf_hdr.seg[seg_idx].len;
 }
 
-static inline void *packet_seg_data(odp_packet_hdr_t *pkt_hdr, uint32_t seg_idx)
+static inline uint8_t *packet_seg_data(odp_packet_hdr_t *pkt_hdr,
```

Comment:
This is for type compatibility:
```
void *odp_packet_seg_data(odp_packet_t pkt, odp_packet_seg_t seg)
{
	odp_packet_hdr_t *pkt_hdr = packet_hdr(pkt);
	uint32_t seg_offset = 0, offset = 0, i;
	uint32_t seg_ndx = _odp_packet_seg_to_ndx(seg);

	while (seg_ndx >= pkt_hdr->buf_hdr.segcount - seg_offset &&
	       pkt_hdr->ref_hdr) {
		seg_ndx -= (pkt_hdr->buf_hdr.segcount - seg_offset);
		offset   = pkt_hdr->ref_offset;
		pkt_hdr  = pkt_hdr->ref_hdr;
		for (i = 0, seg_offset = 0;
		     i < pkt_hdr->buf_hdr.segcount;
		     i++, seg_offset++) {
			if (offset < pkt_hdr->buf_hdr.seg[i].len)
				break;
			offset -= pkt_hdr->buf_hdr.seg[i].len;
		}
	}

	if (odp_unlikely(seg_ndx + seg_offset >= pkt_hdr->buf_hdr.segcount))
		return NULL;

	return packet_seg_data(pkt_hdr, seg_ndx + seg_offset) + offset;
}
```
Without this change the return statement is flagged because of type mismatches.  The previous code didn't need to perform arithmetic on the returned pointer since packets were self-contained before the introduction of references:
```
void *odp_packet_seg_data(odp_packet_t pkt, odp_packet_seg_t seg)
{
	odp_packet_hdr_t *pkt_hdr = packet_hdr(pkt);

	if (odp_unlikely(_odp_packet_seg_to_ndx(seg) >=
			 pkt_hdr->buf_hdr.segcount))
		return NULL;

	return packet_seg_data(pkt_hdr, _odp_packet_seg_to_ndx(seg));
}
```